cmd/satellite/stripe.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+
+	"github.com/prometheus/common/log"
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/private/dbutil"
+	"storj.io/storj/satellite"
+	"storj.io/storj/satellite/payments"
+	"storj.io/storj/satellite/payments/stripecoinpayments"
+	"storj.io/storj/satellite/satellitedb"
+	"storj.io/storj/satellite/satellitedb/dbx"
+)
+
+// Payments is a wrapper around the Account Handling Service
+type Payments struct {
+	Accounts payments.Accounts
+}
+
+// UserData contains the uuid and email of a satellite user
+type UserData struct {
+	ID    uuid.UUID
+	Email string
+}
+
+// generateStripeCustomers creates missing stripe-customers for users in our database
+func generateStripeCustomers(ctx context.Context) error {
+	//Open SatelliteDB for the Payment Service
+	db, err := satellitedb.New(zap.L().Named("db"), runCfg.Database, satellitedb.Options{})
+	if err != nil {
+		return errs.New("error connecting to master database on satellite: %+v", err)
+	}
+	defer func() {
+		err = errs.Combine(err, db.Close())
+	}()
+
+	//Open direct DB connection to execute custom queries
+	driver, source, implementation, err := dbutil.SplitConnStr(runCfg.Database)
+	if err != nil {
+		return err
+	}
+	if implementation != dbutil.Postgres && implementation != dbutil.Cockroach {
+		return errs.New("unsupported driver %q", driver)
+	}
+
+	dbxDB, err := dbx.Open(driver, source)
+	if err != nil {
+		return errs.New("failed opening database via DBX at %q: %v",
+			source, err)
+	}
+	log.Debug("Connected to:", zap.String("db source", source))
+	defer func() {
+		err = errs.Combine(err, dbxDB.Close())
+	}()
+
+	handler, err := setupPayments(zap.L().Named("payments"), db)
+	if err != nil {
+		return err
+	}
+
+	rows, err := dbxDB.Query(ctx, "SELECT id, email FROM users WHERE email NOT IN (SELECT email from stripe_coinpayments")
```

Comment:
shouldn't this be somewhere in DB code and exposed only by interface method?

Reply:
This is only a short lived tool and i am planning on ripping it out after the next release.